rotkehlchen/data_import/utils.py

```diff
@@ -0,0 +1,66 @@
+from abc import ABCMeta, abstractmethod
+from pathlib import Path
+from typing import Any, List, Tuple
+
+from rotkehlchen.accounting.ledger_actions import LedgerAction
+from rotkehlchen.db.dbhandler import DBHandler
+from rotkehlchen.db.ledger_actions import DBLedgerActions
+from rotkehlchen.exchanges.data_structures import AssetMovement, Trade
+
+ROWS_PER_CONTEXT_SWITCH = 500
+
+TRADES_PER_DB_WRITE = 100
+ASSET_MOVEMENTS_PER_DB_WRITE = 100
+LEDGER_ACTIONS_PER_DB_WRITE = 100
+
+
+class BaseExchangeImporter(metaclass=ABCMeta):
+    def __init__(self, db: DBHandler) -> None:
+        self.db = db
+        self.db_ledger = DBLedgerActions(self.db, self.db.msg_aggregator)
+        self._trades: List[Trade] = []
+        self._asset_movements: List[AssetMovement] = []
+        self._ledger_actions: List[LedgerAction] = []
+
+    @abstractmethod
+    def import_csv(self, filepath: Path, **kwargs: Any) -> Tuple[bool, str]:
+        ...
+
+    def add_trade(self, trade: Trade) -> None:
```

Comment:
So this entire concept is not needed here. I have already made multiple changes in the entire code on which you should have rebased which change the way we interact with the DB and you are not using it.

The new way is to create a cursor with a context manager. Either a write cursor if you intend to write data (`with db.user_write() as cursor:` or a read cursor if you intend to read `with db.conn.read_ctx() as cursor:`.

And then you share these cursors across multiple functions.

When the write context manager closes succesfully it writes to the DB and commits.
If something breaks it rolls back.

So I would change this entire approach to follow the new way.

Reply:
Done